script.sql

```diff
@@ -1,147 +1,85 @@
-CREATE DATABASE Lab2;
-USE Lab2;
+-- CREATE
 CREATE TABLE Publisher ( 
-id INT,
-Name varchar(60),
- primary key (id)
+        id INT,
+        name VARCHAR(60),
+        PRIMARY KEY (id)
 );
 
 CREATE TABLE Category (
-id INT,
-name varchar(60),
-parentId INT,
-FOREIGN KEY (parentId) REFERENCES Publisher(id) ON DELETE CASCADE ON UPDATE CASCADE,
-primary key (id)
-
-
-/*FOREIGN KEY  parentId REFERENCES */
+        id INT,
+        name varchar(60),
+        parentId INT,
+        PRIMARY KEY (id),
+        FOREIGN KEY (parentId) REFERENCES Category(id)
 );
-CREATE TABLE App  ( 
-id INT,
-name varchar(60),
-publisherId INT,
-categoryId INT,
-/*FOREIGN KEY (publisherId) REFERENCES Publisher(id) ON DELETE CASCADE ON UPDATE CASCADE,
-FOREIGN KEY (categoryId) REFERENCES category(id) ON DELETE CASCADE ON UPDATE CASCADE, */
-description VARCHAR(255)
 
+CREATE TABLE App  ( 
+        id INT,
+        name VARCHAR(60),
+        publisherId INT,
+        categoryId INT,
+        description VARCHAR(255),
+        PRIMARY KEY (id),
+        FOREIGN KEY (publisherId) REFERENCES Publisher(id) 
+                ON DELETE SET NULL ON UPDATE NO ACTION,
+        FOREIGN KEY (categoryId) REFERENCES Category(id) 
+                ON DELETE SET NULL ON UPDATE NO ACTION
 );
-CREATE TABLE  AppVersion  ( 
-appId INT,
-version char(10),
-
-releaseDate DATETIME,
-rating INT,
-price Decimal(10,2), 
-description VARCHAR(500)
 
+CREATE TABLE  AppVersion  ( 
+        appId INT,
+        version CHAR(10),
+        releaseDate DATETIME,
+        rating INT,
+        price DECIMAL(10,2), 
+        description VARCHAR(500),
+        PRIMARY KEY (appId, version),
+        FOREIGN KEY (appId) REFERENCES App(id)
+                ON DELETE CASCADE ON UPDATE CASCADE
 );
+
 CREATE TABLE AppVersionReview  ( 
-appId INT,
-PRIMARY KEY  (appId),
-version CHAR(10),
-reviewer CHAR(20),
-reviewDate DATETIME,
-rating INT,
-review VARCHAR(1000));
+        appId INT,
+        version CHAR(10),
+        reviewer CHAR(20),
+        reviewDate DATETIME NOT NULL,
+        rating INT,
+        review VARCHAR(1000),
+        PRIMARY KEY (appId, version, reviewer),
+        FOREIGN KEY (appId, version) REFERENCES AppVersion(appId, version)
+                ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+-- INSERT
```

Comment:
Insert statements are all good but because all tuples include every field so there is no need to specify the field names. It looks better right?